src/tests/core.py

```diff
@@ -274,6 +276,33 @@ def test_old_knownnodes_pickle(self):
         finally:
             cleanup(files=('knownnodes.dat',))
 
+    @staticmethod
+    def generate_random_address():
+        import queues
+        streamNumberForAddress = 1
+        queues.addressGeneratorQueue.put((
+            'createRandomAddress', 4, streamNumberForAddress,
+            "test1", 1, "", False))
+
+    def test_add_same_address_twich_in_addressbook(self):
+        """checking same address is added twice in addressbook"""
+        helper_addressbook.insert(label='test1', address=self.addr)
+        self.assertTrue(helper_addressbook.insert(label='test1', address=self.addr))
+
+    def test_is_address_present_in_addressbook(self):
+        """checking is address added in addressbook or not"""
+        helper_addressbook.insert(label='test1', address=self.addr)
```

Comment:
if this is run before `test_add_same_address_twich_in_addressbook`, that one will fail because it's duplicate already. Use a different address, or have both test cases delete the address after they are done.

Reply:
ok